EnhancedMediaPlayer/MediaPlayerViewController.swift

```diff
@@ -0,0 +1,89 @@
+// MediaPlayerViewController.swift
+
+import SwiftUI
+import AVFoundation
+
+public class MediaPlayerViewController: UIHostingController<MediaPlayerView> {
+    private let viewModel: MediaPlayerView.ViewModel
+
+    public init(mediaURL: URL) {
+        self.viewModel = .init(player: .init(url: mediaURL))
+
+        super.init(rootView: .init(viewModel: viewModel))
+
+        setupActions()
+    }
+
+    required dynamic init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    public override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        onLoad()
+    }
+
+    func updateURL(with mediaURL: URL) {
+        viewModel.player = .init(url: mediaURL)
+        onLoad()
+    }
+
+    private func setupActions() {
+        viewModel.onTapAction = { [weak self] control in
+            guard let self else { return }
+            
+            switch control {
+                case .play: self.play()
+                case .pause: self.pause()
+                case .replay: self.replay()
+                case .rewind: self.rewind()
+                case .forward: self.forward()
+            }
+        }
+    }
+    
+    private func play() {
+        self.viewModel.playerState = .playing
+    }
+    
+    private func pause() {
+        self.viewModel.playerState = .paused
+    }
+    
+    private func replay() {
+        // TODO: implement replay
+    }
+    
+    private func rewind() {
+        // TODO: implement rewind
+    }
+    
+    private func forward() {
+        // TODO: implement forward
+    }
+    
+    private func onLoad() {
+        play()
+        observeMediaFinished()
+    }
+    
+    private func observeMediaFinished() {
+        NotificationCenter.default.addObserver(
```

Comment:
Also call `removeObserver` between this lines to avoid dangling references in subsequent `observeMediaFinished` calls